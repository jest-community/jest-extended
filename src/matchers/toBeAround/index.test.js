import matcher from './';

expect.extend(matcher);

describe('.toBeAround', () => {
  'returns true when given number is equal to the rounded expected number';
  test('passes when when given number is equal to the rounded expected number', () => {
    expect(0.25001).toBeAround(0.25, 2);
  });

  test('fails when given number is not equal to the rounded expected number', () => {
    expect(() => expect(0.26).toBeAround(0.25, 2)).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeAround', () => {
  test('when given number is not equal to the rounded expected number', () => {
    expect(0.26).not.toBeAround(0.25, 2);
  });

  test('fails when given number is equal to the rounded expected number', () => {
    expect(() => expect(0.25).not.toBeAround(0.25, 2)).toThrowErrorMatchingSnapshot();
  });
});
