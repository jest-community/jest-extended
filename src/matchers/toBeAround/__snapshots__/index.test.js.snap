// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeAround fails when given number is equal to the rounded expected number 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeAround(</><green>expected</><dim>)</>

expected 0.25 not to be around 0.25"
`;

exports[`.toBeAround fails when given number is not equal to the rounded expected number 1`] = `
"<dim>expect(</><red>received</><dim>).toBeAround(</><green>expected</><dim>)</>

expected 0.26 to be around 0.25"
`;
